#! /bin/sh /usr/share/dpatch/dpatch-run
## gcc-4.1.dpatch by Samuel Mimram <smimram@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make linphone compile with gcc 4.1.

@DPATCH@
diff -urNad linphone-1.3.2~/oRTP/src/sessionset.c linphone-1.3.2/oRTP/src/sessionset.c
--- linphone-1.3.2~/oRTP/src/sessionset.c	2006-02-01 16:30:04.000000000 +0000
+++ linphone-1.3.2/oRTP/src/sessionset.c	2006-03-22 13:58:00.000000000 +0000
@@ -63,9 +63,12 @@
 	uint32_t *mask1,*mask2,*mask3;
 	int i=0;
 	int j,ret=0;
-	mask1=(uint32_t*)&sched_set->rtpset;
-	mask2=(uint32_t*)&user_set->rtpset;
-	mask3=(uint32_t*)&result_set->rtpset;
+	ortp_fd_set *ss=&sched_set->rtpset;
+	ortp_fd_set *us=&user_set->rtpset;
+	ortp_fd_set *rs=&result_set->rtpset;
+	mask1=(uint32_t*)ss;
+	mask2=(uint32_t*)us;
+	mask3=(uint32_t*)rs;
 	while(i<maxs+1){
 		*mask3=(*mask1) & (*mask2);	/* computes the AND between the two masks*/
 		/* and unset the sessions that have been found from the sched_set */
